query/src/main/java/io/keen/client/java/AbsoluteTimeframe.java

```diff
@@ -46,16 +46,20 @@ public AbsoluteTimeframe(String start, String end) {
      */
     @Override
     public Map<String, Object> constructTimeframeArgs() {
-        Map timeframe = null;
+        Map<String, Object> timeframe = null;
+
         if (start != null && end != null) {
-            Map absoluteTimeframe = new HashMap<String, Object>();
+            Map<String, Object> absoluteTimeframe = new HashMap<String, Object>(3);
```

Comment:
I believe we're still compiling for source and target Java 6 compatibility. Maybe moving to something more recent could be added to a list of TODO's for the next major revision? Do you know what the most recent version is that could be targeted given business needs? Are there clients that require running on JRE 1.6? 1.7? Maybe there are implications for Android compatibility? Do you know why Java 6 was originally targeted?